Allow non offset based timezones

### Apache Iceberg Rust version

None

### Describe the bug

I was trying to make PyIceberg rely solely on Iceberg-rust for the partition transforms, but I ran into the following:

```
    def _transform(array: "ArrayLike") -> "ArrayLike":
        if isinstance(array, pa.Array):
            return transform_func(array, *args)
        elif isinstance(array, pa.ChunkedArray):
            result_chunks = []
            for arr in array.iterchunks():
>               result_chunks.append(transform_func(arr, *args))
E               ValueError: Unexpected => Parser error: Invalid timezone "UTC": only offset based timezones supported without chrono-tz feature
```

We initialize the `datetime` as:

```python
datetime(2023, 12, 31, 0, 0, 0, tzinfo=timezone.utc)
```

### To Reproduce

In `test_transform.py`:
```python
def test_year_transform_datetime():
    arr = pa.array([
        datetime(1970, 1, 1, 0, 0, 0, tzinfo=timezone.utc),
        datetime(2000, 1, 1, 0, 0, 0, tzinfo=timezone.utc)
    ])
    result = transform.year(arr)
    expected = pa.array([0, 30], type=pa.int32())
    assert result == expected
```

### Expected behavior

Able to just parse these. I'm not sure if we need to fix this in Arrow, or if we can import `chrono-tz` package.

### Willingness to contribute

I cannot contribute a fix for this bug at this time

--- a/bindings/python/Cargo.toml
+++ b/bindings/python/Cargo.toml
@@ -33,4 +33,4 @@ crate-type = ["cdylib"]
 [dependencies]
 iceberg = { path = "../../crates/iceberg" }
 pyo3 = { version = "0.23.3", features = ["extension-module", "abi3-py39"] }
-arrow = { version = "54.1.0", features = ["pyarrow"] }
+arrow = { version = "54.1.0", features = ["pyarrow", "chrono-tz"] }

--- a/bindings/python/tests/test_transform.py
+++ b/bindings/python/tests/test_transform.py
@@ -15,7 +15,7 @@
 # specific language governing permissions and limitations
 # under the License.
 
-from datetime import date, datetime
+from datetime import date, datetime, timezone
 
 import pyarrow as pa
 import pytest
@@ -58,6 +58,18 @@ def test_bucket_chunked_array():
 
 def test_year_transform():
     arr = pa.array([date(1970, 1, 1), date(2000, 1, 1)])
+    result = transform.year(arr)
+    expected = pa.array([0, 30], type=pa.int32())
+    assert result == expected
+
+
+def test_year_transform_with_tz():
+    arr = pa.array(
+        [
+            datetime(1970, 1, 1, 0, 0, 0, tzinfo=timezone.utc),
+            datetime(2000, 1, 1, 0, 0, 0, tzinfo=timezone.utc),
+        ]
+    )
     result = transform.year(arr)
     expected = pa.array([0, 30], type=pa.int32())
     assert result == expected

